### PR TITLE
Feature extension

### PR DESCRIPTION
**Pull request by TJTorola**
This adds many new features including a timer built into the extension itself, as well as github API requests for the daily curriculum.

### CODE REVIEW

**Comment by TJTorola** [comment]
@composerinteralia The desk hash set up was basically unchanged except that it is now all that is left in the options dialog, you can access it by clicking on the AA logo in the top left.

File: README.md [hunk 1]
```diff
@@ -13,3 +13,105 @@ This is a browser extension to open a new tab with a page of links to commonly u
...
+  request.post({ url: url, form: data(code) }, handleResponse);
+};
+```
+
+4. The Lambda function will make another request to the GH servers, varifying our identity using the private_key and sending the client back the actual API token.
+5. This token is then passed to the curriculum middleware to get the daily curriculum.
+6. Using settings in the `util\settings.js` file and the given day from the AA API, we make a request to GH's API again requesting the readme with today's content.
+7. Then using a functional pipe, we apply a number of regex functions to the readme to parse it for today's content.
+
+### Requirements
```
Comment (composerinteralia) [inline README.md]: Let's make sure to note this somewhere in the [curriculum contribution guidelines](https://github.com/appacademy/instructors/tree/master/curriculum)

**Comment by composerinteralia** [comment]
I get `Cannot read property 'launchWebAuthFlow' of undefined(…)` when I try to authorize with github. What am I missing?

**Comment by composerinteralia** [comment]
We still need a way to set the desk number. Once set, the number should show up in the diamond thing next to the time, and it should also be put into local storage to create the deskHash.

Also, let's put the city id in local storage right away. It's not obvious that you have to specifically click on it to get it there.

**Comment by composerinteralia** [comment]
Another small thing if you have time for it: when I first type the password and press enter it clears the input and closes the modal. The second time it doesn't do either of those things, nor is there any other feedback to indicate the password has updated (although if you check localStorage it does indeed update). I like the first behavior better. Could we have it do that all the time?

**Comment by TJTorola** [comment]
Yeah, so for the first, it just needed to reload some new privileges for the launchWebAuthFlow. To the second, when you have the desks modal up you can click a particular desk to set it. And I'll look into improving the behavior of the password modal.

**Comment by composerinteralia** [comment]
Oh great! I didn't notice the desks were clickable.

**Comment by TJTorola** [comment]
@composerinteralia so, I modified the desk-hash modal functionality. But now that I'm taking another look at it, do we really want to be storing the password for the desk hash calculation on the local state? Would it be better if we didn't store it at all and the desk hash would only be calculated when the password is changed?

I'll get started on storing the cityId right on startup.

**Comment by TJTorola** [comment]
The City id bug is fixed now, should set it in local storage right away

**Comment by composerinteralia** [comment]
Yeah, that would definitely make more sense than putting the password in local storage. We didn't worry about it the past because by the time students learn about local storage we usual aren't as worried about checkins. Thanks again for your hard work on this!

**Comment by TJTorola** [comment]
Done and done. You should double check that the desk hash is still properly working as I had to change it's setter function.

**Comment by composerinteralia** [comment]
Everything seems to be working correctly on my end. Many thanks for the notification!

File: assets/src/components/app/timer/clock_svg/view.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { MINS_IN_SESSION } from 'util/settings';
+
+import { setDashFromSec, setDashFromMin } from 'util/svg_clock';
+
+const SECONDS_RAD = 222; // width = 9
+const BACK_RAD    = 233; // width = 32
+const MINUTES_RAD = 238; // width = 23
+
+const formatMinutes = (minutes) => {
+	if (minutes >= MINS_IN_SESSION) { return MINS_IN_SESSION; }
```
Comment (wolverdude) [inline assets/src/components/app/timer/clock_svg/view.jsx]: tabs

File: README.md [hunk 3]
```diff
@@ -13,3 +13,105 @@ This is a browser extension to open a new tab with a page of links to commonly u
...
+
+### Maintence
+
+The settings file needs to be maintained to point to the correct locations in our curriculum.
+
+__Example Settings__
+
+```
+export const GITHUB = {
+	oauthUrl  : "https://github.com/login/oauth/authorize",
```
Comment (wolverdude) [inline README.md]: tabs

File: assets/src/components/app/timer/handles/view.jsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+import { MINS_IN_SESSION } from 'util/settings';
+const CNT = MINS_IN_SESSION;
+const RADIUS = 240;
+
+const handles = set => {
+	const handleMapper = handle(set);
```
Comment (wolverdude) [inline assets/src/components/app/timer/handles/view.jsx]: tabs

File: assets/src/components/app/header/menu/cities/container.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+const mapStateToProps = state => ({});
+
+const mapDispatchToProps = dispatch => ({
+	set: cityId => dispatch({ type: "SET_CITY_ID", cityId })
+});
+
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(View);
```
Comment (wolverdude) [inline assets/src/components/app/header/menu/cities/container.js]: newline, please

**Comment by wolverdude** [comment]
@composerinteralia: You may merge when ready.

**Pull request merged**